Enable https

johnbradley opened the pull request:
Adds changes patterned after https://github.com/Duke-GCB/D4S2 added to bespin-web.conf.
Enables ssl module. Should redirect http -> https.
Upgrades to bespin-ui that uses ssl for talking to bespin-job-watcher.

Includes changes so we only load sample data once.

johnbradley committed:
ssl setup

Adds changes patterned after https://github.com/Duke-GCB/D4S2 added to bespin-web.conf.
Enables ssl module.
Includes changes so we only load sample data once.

johnbradley committed:
remove debug testing changes

dleehr commented on bespin-web/bespin-web.conf:
```diff
@@ -15,8 +15,21 @@ LogLevel warn
 
-# For now, vhost on port 80. Ultimately this should move to 443 and the 80 will just redirect
+# Always redirect to SSL
+ServerName bespin-dev.gcb.duke.edu
```
Can we replace the 3 literal hostname references with an environment variable?

johnbradley committed:
Adds PassEnv to apache config for server name

Adds instructions to README for creating self signed certs and passing server name.

dleehr commented on bespin-web/scripts/start-apache.sh:
```diff
@@ -5,7 +5,13 @@ python manage.py migrate
 create-lando-user.sh
 create-lando-connection.sh
-load-sample-data.sh
+LOADEDSAMPLEDATA=/var/run/bespin-load.txt
```
What problem is this solving? The fixtures shouldn't be duplicated. At worst, I think it would clobber modifications to the sample data or possibly related objects in the database, but I'm curious.

If it's necessary to only do this once, storing this state in the /var directory of the web container is not a great solution (the web container is otherwise stateless/ephemeral)

johnbradley replied:
The problem is every time you restart bespin-web it loads the fixtures and resets the DukeDS settings. So you have to go in and recreate them. Perhaps removing the two DukeDS them from the fixtures entirely would remedy this.

dleehr replied:
Yeah we should probably remove those from the fixtures then

johnbradley committed:
revert one time load sample data change

johnbradley committed:
Using Rewrite instead of Redirect

In this version of apache Redirect does no variable interpolation.
RewriteRule seems to only support built in environment variables so using SERVER_NAME.

johnbradley commented:
@dleehr Any more thoughts on this one?

dleehr commented:
LGTM
